ClientLib/src/main/java/com/o3dr/android/client/utils/geotag/Geotag.java

```diff
@@ -0,0 +1,232 @@
+                if (Environment.isExternalStorageRemovable(dir)) {
+                    saveDir = dir;
+                    break;
+                }
+            }
+        }
+        return saveDir;
+    }
+
+    private static boolean hasEnoughMemory(String filePath, Collection<File> photos) {
+        StatFs stat = new StatFs(filePath);
+        long freeBytes = stat.getAvailableBytes();
+        long bytesNeeded = 0;
+        for (File photo : photos) {
+            bytesNeeded += photo.length();
+        }
+
+        if (bytesNeeded > freeBytes) {
+            return false;
+        }
+        return true;
+    }
+
+    private static void copyFile(String inputPath, String outputPath) throws IOException {
+        InputStream in = new FileInputStream(inputPath);
+        OutputStream out = new FileOutputStream(outputPath);
+
+        byte[] buffer = new byte[1024];
+        int read;
+        while ((read = in.read(buffer)) != -1) {
+            out.write(buffer, 0, read);
+        }
+        in.close();
+
+        // write the output file (You have now copied the file)
+        out.flush();
+        out.close();
+    }
+
+    private static void sendResult(Handler handler, final GeoTagCallback geoTagCallback, final HashMap<File, File> geoTaggedPhotos,
+                                   final HashMap<File, Exception> failedFiles) {
+        if (geoTagCallback != null) {
+            handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    geoTagCallback.onResult(geoTaggedPhotos, failedFiles);
+                }
+            });
+        }
+    }
+
+    private static void sendFailed(Handler handler, final GeoTagCallback geoTagCallback, final Exception exception) {
+        if (geoTagCallback != null) {
+            handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    geoTagCallback.onFailed(exception);
+                }
+            });
+        }
+    }
+
+    private static void updateExif(TLogParser.Event event, File photoFile) throws IOException {
+        msg_camera_feedback msg = ((msg_camera_feedback) event.getMavLinkMessage());
+        double lat = (double) msg.lat / 10000000;
+        double lng = (double) msg.lng / 10000000;
+        String alt = String.valueOf(msg.alt_msl);
+
+        ExifInterface newExif = new ExifInterface(photoFile.getPath());
+        newExif.setAttribute(ExifInterface.TAG_GPS_LONGITUDE, convertLatLngToDMS(lng));
+        newExif.setAttribute(ExifInterface.TAG_GPS_LATITUDE, convertLatLngToDMS(lat));
+        newExif.setAttribute(ExifInterface.TAG_GPS_LATITUDE_REF, lat < 0 ? "S" : "N");
+        newExif.setAttribute(ExifInterface.TAG_GPS_LONGITUDE_REF, lng < 0 ? "W" : "E");
+        newExif.setAttribute(ExifInterface.TAG_GPS_ALTITUDE, alt);
+        newExif.saveAttributes();
+    }
+
+    private static final String convertLatLngToDMS(double coord) {
+
+        coord = Math.abs(coord);
```

Comment:
@chaviw reusing the same variable makes the logic difficult to follow. Let's strive for clarity whenever possible, and use different appropriately named variables.